go/vt/vtgate/planbuilder/testdata/systemtables_cases.json

```diff
@@ -1,164 +1,146 @@
-        "Query": "select col from information_schema.foo",
-        "Table": "information_schema.foo"
+        "FieldQuery": "select TABLE_NAME from information_schema.`TABLES` where 1 != 1",
+        "Query": "select TABLE_NAME from information_schema.`TABLES`",
+        "Table": "information_schema.`TABLES`"
       }
     }
   },
   {
     "comment": "',' join information_schema",
-    "query": "select a.id,b.id from information_schema.a as a, information_schema.b as b",
-    "plan": {
+    "query": "select a.ENGINE, b.DATA_TYPE from information_schema.TABLES as a, information_schema.COLUMNS as b",
+    "v3-plan": {
       "QueryType": "SELECT",
-      "Original": "select a.id,b.id from information_schema.a as a, information_schema.b as b",
+      "Original": "select a.ENGINE, b.DATA_TYPE from information_schema.TABLES as a, information_schema.COLUMNS as b",
       "Instructions": {
         "OperatorType": "Route",
         "Variant": "DBA",
         "Keyspace": {
           "Name": "main",
           "Sharded": false
         },
-        "FieldQuery": "select a.id, b.id from information_schema.a as a, information_schema.b as b where 1 != 1",
-        "Query": "select a.id, b.id from information_schema.a as a, information_schema.b as b",
-        "Table": "information_schema.a, information_schema.b"
+        "FieldQuery": "select a.`ENGINE`, b.DATA_TYPE from information_schema.`TABLES` as a, information_schema.`COLUMNS` as b where 1 != 1",
+        "Query": "select a.`ENGINE`, b.DATA_TYPE from information_schema.`TABLES` as a, information_schema.`COLUMNS` as b",
+        "Table": "information_schema.`TABLES`, information_schema.`COLUMNS`"
       }
-    }
-  },
-  {
-    "comment": "information schema query that uses table_schema",
-    "query": "select column_name from information_schema.columns where table_schema = (select schema())",
-    "plan": {
+    },
+    "gen4-plan": {
       "QueryType": "SELECT",
-      "Original": "select column_name from information_schema.columns where table_schema = (select schema())",
+      "Original": "select a.ENGINE, b.DATA_TYPE from information_schema.TABLES as a, information_schema.COLUMNS as b",
       "Instructions": {
         "OperatorType": "Route",
         "Variant": "DBA",
         "Keyspace": {
           "Name": "main",
           "Sharded": false
         },
-        "FieldQuery": "select column_name from information_schema.`columns` where 1 != 1",
-        "Query": "select column_name from information_schema.`columns` where table_schema = schema()",
-        "Table": "information_schema.`columns`"
+        "FieldQuery": "select a.`ENGINE`, b.DATA_TYPE from information_schema.`TABLES` as a, information_schema.`COLUMNS` as b where 1 != 1",
+        "Query": "select a.`ENGINE`, b.DATA_TYPE from information_schema.`TABLES` as a, information_schema.`COLUMNS` as b",
+        "Table": "information_schema.`COLUMNS`, information_schema.`TABLES`"
```

Comment:
any idea why the two tables are in inverse orders between the Gen4 and V3 plans? Is that flakky?

Reply:
don't know why. I don't think it's flaky, it's never failed locally even after many runs